src/v/model/adl_serde.h

```diff
@@ -178,6 +178,9 @@ adl<model::record_batch_header>::parse_from(Parser& in) {
     using tmstmp_t = model::timestamp::type;
     auto first = model::timestamp(adl<tmstmp_t>{}.from(in));
     auto max = model::timestamp(adl<tmstmp_t>{}.from(in));
+    if (max < first) {
+        max = first;
+    }
```

Comment:
this strikes me as something that should be done by the caller that is deserializing a record batch rather than at the lowest level in serialization routine.

Reply:
good point, got rid of it